classes/that.sc

```diff
@@ -21,51 +21,54 @@ That {
 		all = ();
 	}
 
-	*new {|name, input, analyzerFunction, callback|
+	*new { |name, input, analyzerFunction, callback|
 		var res = all.at(name);
 		if(res.isNil, {
-			if(analyzerFunction.isNil, {
-				Error("Please provide an analyzerFunction").throw;
-			});
-			callback = callback ? {}; // enforce a function as callback and not allow nil
-			res = super.newCopyArgs(name, input, analyzerFunction, callback).init;
+			res = super.new.init;
 			all[name] = res;
-		}, {
-			input !? { res.input = input };
-			analyzerFunction !? { res.analyzerFunction = analyzerFunction };
-			callback !? { res.callback = callback };
 		});
+		res.setFunctions(input, analyzerFunction, callback);
 		^res
 	}
 
 	init {
 		defName = "that_%".format(name).asSymbol;
 		oscChannelName = "/that/%".format(name);
-		this.input = input;
+		analyzer = Ndef(defName);
+		oscdef = OSCdef(defName);
 	}
 
 	clear {
-		OSCdef(defName).free;
-		Ndef(defName).clear;
+		analyzer.free;
+		oscdef.clear;
 		all[name] = nil;
 	}
 
-	input_ {|newInput|
+	input_ { |newInput|
 		input = newInput;
-		this.prUpdateDefs();
+		this.prUpdateDefs;
```

Comment:
I personally prefer to use `()` on function calls as I found this implicitness really confusing when I started sclang (especially considering that sclang also allows for higher order functions as well and all other languages I use don't just call the function in this case) and for me it leads to more readability b/c it removes one level of implicitness as it could just be a leftover of a parameter access.
It is also more consistent with the writing if the function needs parameters.

But at the same time, we can switch to this writing as well.

Reply:
Yes, I understand.

Nevertheless: Better switch, because this is idiomatic. 
One shouldn't think "function call" but message passing. A message with no arguments is just a word: 
```supercollider
{ (0..100).clump(3).scramble.flat }.dup.flop
```

Reply:
I think if your sc-native than this style of code is more common but especially on chained calls I find this extremely confusing as I can not separate between property and method. Also what happens if you need to pass a function as a parameter: does this get called as well (no, not yet, only when you call it) but the semantics help a lot in such cases to understand what one is passing around in such cases.

But as this is SC code one just has to accept it and start getting used to it I think.
For myself I really try to code as explicit as possible in SC as the implicitness of SC just took some of my lifetime already (fav of mine is `nil.size` is `0` combined with the fact that every unset variable is initiated as nil).